test/functional/wallet_resendwallettransactions.py

```diff
@@ -74,6 +79,41 @@ def run_test(self):
         node.setmocktime(now + 36 * 60 * 60 + 600)
         peer_second.wait_for_broadcast([txid])
 
+        self.log.info("Chain of unconfirmed not-in-mempool txs are rebroadcast")
```

Comment:
Question: Is this test meant to fail without the first commit?

Reply:
It's passing for me on master :thinking:

Reply:
Since master has `mapWallet` as a `std::unsorted_map`, this will now only fail randomly on it. It should fail half of the time, on average. I don't know how we can guarantee that the child will come before the parent in an unsorted map.

Reply:
I've removed the "make a spend that will come before" and updated the comment to explain that.

Reply:
Hm, I see what you're saying about `std::unsorted_map` and it makes sense to me, but I've run the test >20 times and haven't gotten a failure yet...

Reply:
Found a bug in the test. It should fail intermittently on master now

Reply:
Aha, fails intermittently on master for me now :) thanks!

Reply:
I'm seeing the test fail intermittently on commit [e40ddf3](https://github.com/bitcoin/bitcoin/pull/25768/commits/e40ddf36bed81bdf28d386eb961c9ed22b69e207) does that sound right or should the test only fail on the master branch

Reply:
It should only fail on master. It should not fail on this PR. Make sure that you've compiled this PR before running the test.

Reply:
hmm I think I've compiled it correctly (still new to the repo so might be doing it incorrectly), this is what I did

```
git fetch origin pull/25768/head:PR25768
git checkout  PR25768
./autogen.sh
./configure
make clean
make -j "$(($(sysctl -n hw.physicalcpu)+1))"
./test/functional/wallet_resendwallettransactions.py
```

This is the error I see after I run the test, sometimes it passes

```
2022-08-18T20:38:25.914000Z TestFramework (INFO): Initializing test directory /var/folders/9g/cvx014yx4dq5lwl_x5zvn_j80000gn/T/bitcoin_func_test_43xe3o56
2022-08-18T20:38:28.879000Z TestFramework (INFO): Create a new transaction and wait until it's broadcast
2022-08-18T20:38:29.627000Z TestFramework (INFO): Create a block
2022-08-18T20:38:29.634000Z TestFramework (INFO): Bump time & check that transaction is rebroadcast
2022-08-18T20:38:29.905000Z TestFramework (INFO): Chain of unconfirmed not-in-mempool txs are rebroadcast
2022-08-18T20:38:31.037000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py", line 133, in main
    self.run_test()
  File "/Users/kevkevin/DEVDIR/bitcoin/./test/functional/wallet_resendwallettransactions.py", line 107, in run_test
    assert_raises_rpc_error(-5, "Transaction not in mempool", node.getmempoolentry, txid)
  File "/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/util.py", line 126, in assert_raises_rpc_error
    assert try_rpc(code, message, fun, *args, **kwds), "No exception raised"
AssertionError: No exception raised
2022-08-18T20:38:31.089000Z TestFramework (INFO): Stopping nodes
2022-08-18T20:38:31.508000Z TestFramework (WARNING): Not cleaning up dir /var/folders/9g/cvx014yx4dq5lwl_x5zvn_j80000gn/T/bitcoin_func_test_43xe3o56
2022-08-18T20:38:31.508000Z TestFramework (ERROR): Test failed. Test logging available at /var/folders/9g/cvx014yx4dq5lwl_x5zvn_j80000gn/T/bitcoin_func_test_43xe3o56/test_framework.log
2022-08-18T20:38:31.508000Z TestFramework (ERROR):
2022-08-18T20:38:31.509000Z TestFramework (ERROR): Hint: Call /Users/kevkevin/DEVDIR/bitcoin/test/functional/combine_logs.py '/var/folders/9g/cvx014yx4dq5lwl_x5zvn_j80000gn/T/bitcoin_func_test_43xe3o56' to consolidate all logs
2022-08-18T20:38:31.509000Z TestFramework (ERROR):
2022-08-18T20:38:31.509000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
2022-08-18T20:38:31.509000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues
2022-08-18T20:38:31.509000Z TestFramework (ERROR):
```

running on MacOs Monterey Version 12.3.1

Reply:
I'm not seeing that error even when running on a loop.

Reply:
> git checkout  PR25768

Can you check that `git log -1` prints the correct hash, or alternatively just checkout the correct hash directly? Also, make sure that the working tree is clean.

```
git fetch origin e40ddf36bed81bdf28d386eb961c9ed22b69e207
git checkout e40ddf36bed81bdf28d386eb961c9ed22b69e207
git reset --hard HEAD # warning!! wipes unstaged changes in the tree if there still are some

Reply:
still getting the error and I am able to see I'm on the correct commit with `git log -1` and tried thee commands you posted aswell, I'll try and dig deeper to see what may be the issue

Reply:
not exactly sure what happened but after rebuilding master then rebuilding this PR's branch twice the test passes now sorry for any confusion